lib/api/src/rest/conversions.rs

```diff
@@ -19,6 +18,13 @@ impl From<Vector> for segment::data_types::vectors::Vector {
         match value {
             Vector::Dense(vector) => segment::data_types::vectors::Vector::Dense(vector),
             Vector::Sparse(vector) => segment::data_types::vectors::Vector::Sparse(vector),
+            Vector::MultiDense(vector) => {
+                // the REST vectors have been validated already
+                // we can use an internal constructor
+                segment::data_types::vectors::Vector::MultiDense(
+                    segment::data_types::vectors::MultiDenseVector::new_validated(vector),
```

Comment:
Without this trick, I would be force to use a `TryFrom` here and propagate it to all callsites of this conversions.
The change was massive so I decided to assume that the input vectors are valid at this point based on my tests.